### PR TITLE
[Events] Sandbox for new Events Layout and Chat POC

### PR DESCRIPTION
**Pull request by rkdavidson**
# About
Adds a sandbox version of the existing Event page, and integrates a crude chat POC along with general layout/structural changes to support it.

**TL;DR on the Sandbox configuration**
Files named `.sandbox.js` in the **web-app** are loaded _only in the beta_ environment.
This is separate altogether from the feature flagging system, and is a simple way to maintain old and new versions of a page without worrying about end-user visibility in the mean time.

After this PR is merged, the current production CF events page should remain unchanged — but the same page on beta.christfellowship.church should start using the new `.sandbox.js` versions of components.

# Tradeoffs/Caveats
**Responsive handling** and the mobile view in general were ignored. No real effort was spent massaging padding/spacing details or what happens on smaller viewports.

**Stream.io chat component styling** is still a bit of an unknown. I added a [stream-chat-react-overrides.css](https://github.com/christfellowshipchurch/web-app/pull/149/files#diff-c0eed21003356fc2d1e69876c214907d) file for now, to allow hard coding the height for scaffolding purposes. This solution is far from ideal, but just needs a closer look.

**User authentication** and general Stream client management is _very_ crude here. Next PR will have proper initialization and management of Stream client, with loading/connection state handlers/token fetching/etc. (Bulk of this is on Caleb's branch already, and/or parallels mobile app).

# Test Instructions
You'll need to update your local `.env` file to include a new `REACT_APP_STREAM_API_KEY` value.
The stream.io credentials/keys should match what your local API uses.

1. Start web app and api locally
1. Go to an event page and verify you can see the chat off to the side and general WIP presentation (screenshots below)
1. All messages you send will be as **Yoda**, whether you're logged in or out
1. In the PR CI/CD area, find the **netlify/deploy-preview** jobs and smoke test that the new development page only shows on beta.christfellowship.church

# Screenshots

|Large|Medium|Small|
|---|---|---|
|CleanShot 2020-08-28 at 15 22 48@2x|CleanShot 2020-08-28 at 15 23 09@2x|CleanShot 2020-08-28 at 15 23 18@2x|

# Tests
- [ ] needed snapshots updated
- [ ] mobile tested
- [ ] latest update from master

### CODE REVIEW

**Comment by rkdavidson** [comment]
@hellogerard 
1. Good catch, I fixed that along with re-lint-fixing these files now that the lint PR got merged to `master`. Commit: 58297bc

2. **TL;DR** there's not a _ton_ of value in using `styled-components`, since the overhead of recreating the "foundation" bootstrap provides with it's grid system, responsive handling, etc is very clearly not worth it right now.

There may be some value in at least setting it up, and using it to help guide the structure of components and view hierarchy — since it does enforce a certain approach/thinking. RE: Stream component styling, I thought the same things.
@burkeshartsis had a good point that it may be worth it, even if ultimately we're just using it to specify only `classnames` and no custom "real" css.

Current Plan A is recreate ultra lean/minimal components if possible, borrowing heavily from their internal state management etc...
Plan B, we can gratuitously override their compiled CSS classes for the MVP 🤷‍♂️. We could make it a "cleaner hack" by throwing a top level specificity thing, to isolate the overrides to just Events work... something like `.EventDetailPage .str-react-chat__whatever` instead of just `.str-react-chat__whatever`.

**Comment by hellogerard** [comment]
> TL;DR there's not a ton of value in using styled-components

I hear ya. 

> Plan B, we can gratuitously override their compiled CSS classes for the MVP 🤷‍♂️. We could make it a "cleaner hack" by throwing a top level specificity thing

This is what I was thinking. Not to start completely from scratch, but to augment. I noticed this in the docs:

https://styled-components.com/docs/advanced#existing-css
